packages/lsk/source/ledger/ledger.transport.ts

```diff
@@ -0,0 +1,265 @@
+		let inputBuffer: Buffer;
+		if (Array.isArray(hexData)) {
+			inputBuffer = Buffer.concat(
+				hexData.map((item) => {
+					if (typeof item === "string") {
+						return Buffer.from(item, "hex");
+					} else if (typeof item === "number") {
+						return Buffer.alloc(1).fill(item);
+					}
+					return item;
+				}),
+			);
+		} else if (typeof hexData === "string") {
+			inputBuffer = Buffer.from(hexData, "hex");
+		} else if (typeof hexData === "number") {
+			inputBuffer = Buffer.alloc(1).fill(hexData);
+		} else {
+			inputBuffer = hexData;
+		}
+
+		// Send start comm packet
+		const startCommBuffer = Buffer.alloc(2);
+		startCommBuffer.writeUInt16BE(inputBuffer.length, 0);
+
+		await this.transport.send(0xe0, 89, 0, 0, startCommBuffer);
+
+		// Calculate number of chunks to send.
+		const chunkDataSize = this.chunkSize;
+		const nChunks = Math.ceil(inputBuffer.length / chunkDataSize);
+
+		let previousCRC = 0;
+		for (let index = 0; index < nChunks; index++) {
+			const dataSize = Math.min(inputBuffer.length, (index + 1) * chunkDataSize) - index * chunkDataSize;
+
+			// copy chunk data
+			const dataBuffer = inputBuffer.slice(index * chunkDataSize, index * chunkDataSize + dataSize);
+
+			let resBuffer;
+			try {
+				resBuffer = await this.transport.send(0xe0, 90, 0, 0, dataBuffer);
+			} catch (error) {
+				if (error.message.includes("0x6867")) {
+					throw new Error("Payload too big for Lisk Ledger implementation");
+				} else {
+					throw error;
+				}
+			}
+			const [currentCRC, previousCRCLedger] = this.decomposeResponse(resBuffer);
+			const crc = getCrc16(dataBuffer);
+			const receivedCRC = currentCRC.readUInt16LE(0);
+
+			if (crc !== receivedCRC) {
+				throw new Error("Something went wrong during CRC validation");
+			}
+
+			if (previousCRCLedger.readUInt16LE(0) !== previousCRC) {
+				throw new Error("Prev CRC is not valid");
+			}
+
+			previousCRC = crc;
+		}
+		// Close comm flow.
+		const resBuf = await this.transport.send(0xe0, 91, 0, 0);
+
+		return this.decomposeResponse(resBuf);
+	}
+
+	/**
+	 * Raw sign protocol utility. It will handle signature of both msg and txs.
+	 * @param {number} signType type of signature. 0x05 for txs, 0x06 for messages.
+	 * @param {LedgerAccount|Buffer} account acount or bip32 buffer
+	 * @param {Buffer} buff buffer to sign
+	 * @param {boolean} hasRequesterPKey if it has a requesterpublickey (used only in tx signing mode)
+	 * @returns {Promise<Buffer>} the signature
+	 */
+	private async sign(signType: number, account: LedgerAccount | Buffer, buff: Buffer): Promise<Buffer> {
+		const pathBuf = Buffer.isBuffer(account) ? account : account.derivePath();
+		const buffLength = Buffer.alloc(2);
+		buffLength.writeUInt16BE(buff.length, 0);
+		const arguments_ = await this.exchange([
+			signType, // sign
+			// Bip32
+			pathBuf.length / 4,
+			pathBuf,
+			// headers
+			buffLength,
+			// data
+			buff,
+		]);
+		const [signature] = arguments_;
+		return signature;
```

Comment:
```suggestion
		return arguments_[0];
```

Should be the same as the array destructure?